airflow/utils/task_group.py

```diff
@@ -454,6 +455,26 @@ def topological_sort(self, _include_subdag_tasks: bool = False):
 
         return graph_sorted
 
+    def get_task_dict(self) -> dict[str, AbstractOperator]:
+        """Returns a flat dictionary of task_id: AbstractOperator"""
+        task_map: dict[str, AbstractOperator] = {}
+        groups_to_visit = [self]
+
+        while groups_to_visit:
+            visiting = groups_to_visit.pop(0)
+
+            for child in visiting.children.values():
+                if isinstance(child, AbstractOperator):
+                    task_map[child.task_id] = child
+                elif isinstance(child, TaskGroup):
+                    groups_to_visit.append(child)
+                else:
+                    raise ValueError(
+                        f"Encountered a DAGNode that is not a TaskGroup or an AbstractOperator: {type(child)}"
+                    )
+
+        return task_map
```

Comment:
Hmm, this feels like a pretty expensive way for what is needed. Unless we actually need to dict, using an iterator is likely better (for this particular feature):

```python
def iter_tasks(self) -> Iterator[AbstractOperator]:
    groups_to_visit = [self]
    while groups_to_visit:
        visiting = groups_to_visit.pop(0)
        for child in visiting.children.values():
            if isinstance(child, AbstractOperator):
                yield child
            elif isinstance(child, TaskGroup):
                groups_to_visit.append(child)
            else:
                raise ValueError(f"Encountered unknown DAGNode type: {type(child)}")
```

Reply:
True, for this specific use case, this is much better. Thanks !

Done